Kickstarter-iOS/Views/CommentComposerView.swift

```diff
@@ -0,0 +1,212 @@
+import Library
+import Prelude
+import UIKit
+
+private enum Layout {
+  enum Avatar {
+    static let diameter: CGFloat = 44.0
+  }
+
+  enum Border {
+    static let height: CGFloat = 1.0
+  }
+}
+
+protocol CommentComposerViewDelegate: AnyObject {
+  func commentComposerView(_ view: CommentComposerView, didSubmitText text: String)
+}
+
+final class CommentComposerView: UIView {
+  // MARK: - Properties
+
+  lazy var avatarImageView = {
+    CircleAvatarImageView(frame: .zero)
+      |> \.translatesAutoresizingMaskIntoConstraints .~ false
+  }()
+
+  private lazy var inputContainerView: CommentInputContainerView = {
+    CommentInputContainerView()
+      |> \.translatesAutoresizingMaskIntoConstraints .~ false
+  }()
+
+  private lazy var onlyBackersLabel: UILabel = {
+    UILabel(frame: .zero)
+      |> \.translatesAutoresizingMaskIntoConstraints .~ false
+  }()
+
+  private let rootStackView: UIStackView = { UIStackView(frame: .zero) }()
+  private let topBorderView = {
+    UIView(frame: .zero)
+      |> \.translatesAutoresizingMaskIntoConstraints .~ false
+  }()
+
+  weak var delegate: CommentComposerViewDelegate?
+  private let viewModel: CommentComposerViewModelType = CommentComposerViewModel()
+
+  // MARK: - Lifecycle
+
+  override init(frame: CGRect) {
+    super.init(frame: frame)
+
+    self.bindViewModel()
+    self.configureViews()
+    self.setupConstraints()
+  }
+
+  required init?(coder _: NSCoder) {
+    fatalError("init(coder:) has not been implemented")
+  }
+
+  override var intrinsicContentSize: CGSize {
+    return .zero
+  }
+
+  // MARK: - Configuration
+
+  public func configure(with data: CommentComposerViewData) {
+    self.viewModel.inputs.configure(with: data)
+  }
+
+  /*
+   Call this to clear body text.
+   */
+  public func clearOnSuccess() {
+    self.viewModel.inputs.bodyTextDidChange("")
```

Comment:
This would be a separate VM input, `clearOnSuccess()` we can address then when it's implemented but then let's add a `TODO` here.

Reply:
Okay